File: cesium-simulation-platform/cesium1/server/routes/geology.js
const { Router } = require('express')
const db = require('../db')
const router = Router()

router.get('/orebodies', (_req, res) => {
  try { res.json({ ok: true, data: db.prepare('SELECT * FROM geology_orebodies').all() }) }
  catch (err) { res.status(500).json({ ok: false, message: err.message }) }
})

router.get('/stats', (_req, res) => {
  try {
    const rows = db.prepare('SELECT stat_key, stat_value FROM geology_stats').all()
    const stats = {}
    for (const row of rows) stats[row.stat_key] = row.stat_value
    res.json({ ok: true, data: stats })
  } catch (err) { res.status(500).json({ ok: false, message: err.message }) }
})

router.get('/full', (_req, res) => {
  try {
    const orebodies = db.prepare('SELECT * FROM geology_orebodies').all()
    const rows = db.prepare('SELECT stat_key, stat_value FROM geology_stats').all()
    const stats = {}
    for (const row of rows) stats[row.stat_key] = row.stat_value
    res.json({ ok: true, data: { orebodies, stats } })
  } catch (err) { res.status(500).json({ ok: false, message: err.message }) }
})

module.exports = router

File: cesium-simulation-platform/backend-py/app/routes/geology.py
from fastapi import APIRouter, Depends
from pymysql import Connection
from app.database import get_db

router = APIRouter(prefix="/api/geology", redirect_slashes=False, tags=["地质分析"])


@router.get("/stats")
def get_geology_stats(db: Connection = Depends(get_db)):
    with db.cursor() as cursor:
        cursor.execute("SELECT stat_key, stat_value FROM geology_stats")
        rows = cursor.fetchall()
    stats = {}
    for row in rows:
        stats[row["stat_key"]] = row["stat_value"]
    return {"code": 0, "data": stats}



